### PR TITLE
murdock: improve error message should "info-boards-supported" fail

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

~~DON'T MERGE until test commit has been removed.~~

Sometimes a makefile breaks in a way that makes "make info-boards-supported" fail.
Currently, this results in a failed "compile job" named "compile/appname/makefile_broken", which is confusing.

This PR:

- adds a commit that propagates the error through to the recently introduced toolchain support
- improves the error message when a makefile error is detected
- actually executes the probably failed makefile command so it's output ends up in the failed job.

### Testing procedure

- check the output with the commits enabled
- make sure the normal build still works as expected
- possibly compare the output with just the test commit.

### Issues/PRs references

#11122

### CODE REVIEW

**Comment by maribu** [comment]
Regarding testing: The output with a broken Makefile is as expected: The error message clearly states what the issue is and how to debug on the developers machine. I'd say wait for @smlng feedback and then squash and check if Murdock still works as expected with no broken Makefiles.

File: .murdock [hunk 1]
```diff
@@ -137,7 +147,12 @@ compile() {
     local board=$(echo $2 | cut -f 1 -d':')
     local toolchain=$(echo $2 | cut -f 2 -d':')
 
-    [ "$board" = "makefile_broken" ] && error "$0: Makefile in \"$appdir\" seems to be broken!"
+    [ "$board" = "makefile_broken" ] && {
+        echo "$0: There seems to be a problem in \"$appdir\" while getting supported boards!"
+        echo "$0: testing \"make -C$appdir make-info-boards-supported\"..."
+        make -C$appdir make-info-boards-supported && echo "$0: success. no idea what's wrong." || echo "$0: failed!"
```
Comment (smlng) [inline .murdock]: is this right? I cannot find a target `make-info-boards-supported` just `info-boards-supported`:

```
git grep make-info-boards-supported

```

but 
```
git grep info-boards-supported
.murdock:    local boards="$(make --no-print-directory -C$appdir info-boards-supported 2>/dev/null || echo broken)"
Makefile.include:GLOBAL_GOALS += buildtest info-boards-supported info-boards-features-missing info-buildsizes info-buildsizes-diff
dist/tools/compile_and_test_for_board/compile_and_test_for_board.py:        cmd = ['info-boards-supported']
makefiles/info-global.inc.mk:info-boards-supported:
makefiles/info.inc.mk:.PHONY: info-objsize info-buildsizes info-build info-boards-supported \
makefiles/info.inc.mk:  @echo $$($(MAKE) info-boards-supported)
```

Comment (kaspar030) [reply .murdock]: uff, well spotted. this didn't show up in my test case as that broke the syntax of the makefile, not even trying to build the (non-existing) target "make-info-boards-supported".

fixed.

**Comment by maribu** [comment]
OK. I'd suggest to now squash once @smlng agrees and check if Murdock can build successfully without the broken Makefile added as test case.

**Comment by smlng** [comment]
yep, please squash!

**Comment by kaspar030** [comment]
- squashed

**Comment by kaspar030** [comment]
@smlng only an ACK missing here.

File: .murdock [hunk 2]
```diff
@@ -137,7 +144,12 @@ compile() {
     local board=$(echo $2 | cut -f 1 -d':')
     local toolchain=$(echo $2 | cut -f 2 -d':')
 
-    [ "$board" = "makefile_broken" ] && error "$0: Makefile in \"$appdir\" seems to be broken!"
+    [ "$board" = "makefile_broken" ] && {
+        echo "$0: There seems to be a problem in \"$appdir\" while getting supported boards!"
+        echo "$0: testing \"make -C$appdir make info-boards-supported\"..."
```
Comment (maribu) [inline .murdock]: Shouldn't it be `make -C$appdir info-boards-supported`? (Duplicated `make` here)

Comment (maribu) [reply .murdock]: Btw: Feel free to squash right away.

Comment (kaspar030) [reply .murdock]: fixed & squashed. thx!

**Pull request merged**